test/services/ReviewServiceSpec.scala

```diff
@@ -588,6 +639,24 @@ class ReviewServiceSpec extends UnitSpec with MockFactory with ReviewData with A
           }
         }
       }
+      "the updateReview fails to update the details" should {
+        "return a not found response" in new Test {
+
+          val expected: RequestOutcome[Unit] = Left(Errors(NotFoundError))
+
+          MockApprovalRepository
+            .getById("validId")
+            .returns(Future.successful(Right(factCheckProcess)))
+
+          MockApprovalProcessReviewRepository
+            .updateReview("validId", factCheckProcess.version, ReviewTypeFactCheck, statusChange2iReviewInfo.userId, statusChange2iReviewInfo.status)
+            .returns(Future.successful(Left(Errors(NotFoundError))))
```

Comment:
Check changeStatus is not called.

Reply:
if changeStatus was called, wouldn't test fail with unexpected call?

Reply:
Actually you might be right, but you need to be careful. The problem could be the other way around; you define a mock call that isn't used. I'd expect the test to fail but I'm not sure it does.

Reply:
updated